src/cpp/session/modules/SessionFind.cpp

```diff
@@ -487,6 +489,42 @@ class GrepOperation : public boost::enable_shared_from_this<GrepOperation>
       }
    }
 
+   Error setPermissions(const std::string& filePath, boost::filesystem::perms permissions)
+   {
+      boost::filesystem::path path(filePath);
```

Comment:
When constructing a `boost::filesystem::path` from a `std::string`, some extra work is needed on Windows; see: https://github.com/rstudio/rstudio/blob/6a45dc8e1fd19a093a68bd5b0a533c34b10293b0/src/cpp/shared_core/FilePath.cpp#L224-L246

This is one reason that we almost never use raw `boost::filesystem::path` objects and instead always use our `FilePath` wrapper. You might consider making these set/get permissions methods part of the FilePath object itself (which stores a boost filesystem path internally).

Reply:
At first I did have this as part of `FilePath`, but I felt weird having our wrapper functions directly pass and return `boost::filesystem::perms` without its own wrapper and implementing that seems like more work will little payoff. What are your thoughts on this?

Reply:
That's a good instinct; FilePath should abstract boost::filesystem completely so we shouldn't return filesystem internals like `perms`.

I think we could greatly simplify this by skipping all permissions-related code on Windows. The only permission supported by `boost::filesystem::perms` on Windows is write permission... which is a tautology here since we just wrote to the file! https://www.boost.org/doc/libs/1_66_0/libs/filesystem/doc/reference.html#Enum-perms

We could also simplify it further by using `::stat` and `::chmod` directly instead of using the Boost wrappers around them (which would remove the need to handle exceptions and path wrappers); but I'll leave that up to your discretion.

Reply:
Hi Jonathan, at this point I can't think of a strong argument to use `::stat` and `::chmod` over the boost permissions. I updated the code so these functions are not declared or called in Windows. Thanks!

Reply:
Could we add an update/set permissions method on FilePath to resolve this?

Reply:
Maria pointed out that `FileMode` already has functions I can use so expect another update later today :)

Reply:
IIRC the FileMode enums do not cover all the possible permissions, so I don't think you can use them in this case.

Reply:
Ah you are correct. In that case, I'll leave the code alone and this is ready for review again.